crates/trie/parallel/src/proof_task.rs

```diff
@@ -112,34 +118,78 @@ enum StorageWorkerJob {
 /// # Shutdown
 ///
 /// Worker shuts down when the crossbeam channel closes (all senders dropped).
-fn storage_worker_loop<Tx>(
-    proof_tx: ProofTaskTx<Tx>,
+fn storage_worker_loop<Factory>(
+    view: ConsistentDbView<Factory>,
+    task_ctx: ProofTaskCtx,
     work_rx: CrossbeamReceiver<StorageWorkerJob>,
     worker_id: usize,
     #[cfg(feature = "metrics")] metrics: ProofTaskTrieMetrics,
 ) where
-    Tx: DbTx,
+    Factory: DatabaseProviderFactory<Provider: BlockReader>,
 {
     tracing::debug!(
         target: "trie::proof_task",
         worker_id,
-        "Storage worker started"
+        "Storage worker spawned"
     );
 
-    // Create factories once at worker startup to avoid recreation overhead.
-    let (trie_cursor_factory, hashed_cursor_factory) = proof_tx.create_factories();
-
-    // Create blinded provider factory once for all blinded node requests
-    let blinded_provider_factory = ProofTrieNodeProviderFactory::new(
-        trie_cursor_factory.clone(),
-        hashed_cursor_factory.clone(),
-        proof_tx.task_ctx.prefix_sets.clone(),
-    );
+    // Lazy initialization - transaction created on first job
+    let mut proof_tx: Option<ProofTaskTx<_>> = None;
+    let mut trie_cursor_factory: Option<_> = None;
+    let mut hashed_cursor_factory: Option<_> = None;
+    let mut blinded_provider_factory: Option<_> = None;
 
     let mut storage_proofs_processed = 0u64;
     let mut storage_nodes_processed = 0u64;
 
     while let Ok(job) = work_rx.recv() {
+        // Perform lazy initialization on first job
+        if proof_tx.is_none() {
```

Comment:
can we initialize it not lazily, but eagerly, to not spend time on it when the first job is received?

Reply:
yeah makes sense